frontend/packages/ux-editor/src/components/config/editModal/EditDataModelBinding/EditBinding/EditBinding.test.tsx

```diff
@@ -0,0 +1,424 @@
+    dataType: defaultDataModel,
+  },
+};
+
+type MockedParentComponentProps = EditBindingProps;
+
+const MockedParentComponent = (props: MockedParentComponentProps) => {
+  const [newInternalBindingFormat, setNewInternalBindingFormat] =
+    React.useState<InternalBindingFormat>(props.internalBindingFormat);
+  return (
+    <EditBinding
+      {...props}
+      handleComponentChange={(formItem) => {
+        setNewInternalBindingFormat((prev) => ({
+          ...prev,
+          field: formItem.dataModelBindings[defaultBindingKey],
+        }));
+      }}
+      internalBindingFormat={newInternalBindingFormat}
+    />
+  );
+};
+
+type RenderEditBinding = {
+  editBindingProps?: EditBindingProps;
+  queryClient?: ReturnType<typeof createQueryClientMock>;
+  queries?: Partial<ServicesContextProps>;
+  shouldRenderWithMockedParent?: boolean;
+};
+
+const renderEditBinding = ({
+  editBindingProps = defaultEditBinding,
+  queryClient = createQueryClientMock(),
+  queries,
+  shouldRenderWithMockedParent = false,
+}: RenderEditBinding) => {
+  return {
+    ...renderWithProviders(
+      shouldRenderWithMockedParent ? (
+        <MockedParentComponent {...editBindingProps} />
+      ) : (
+        <EditBinding {...editBindingProps} />
+      ),
+      {
+        queries: { ...queries },
+        queryClient,
+      },
+    ),
+  };
+};
+
+const getAppMetadataModelIdsMock = jest
+  .fn()
+  .mockImplementation(() => Promise.resolve([defaultDataModel, secondDataModel]));
+const getDataModelMetadataMock = jest
+  .fn()
+  .mockImplementation(() => Promise.resolve(dataModelMetadataResponseMock));
+
+describe('EditBinding', () => {
+  it('should render loading spinner', async () => {
+    renderEditBinding({});
+
+    const loadingSpinnerTitle = textMock('ux_editor.modal_properties_loading');
+
+    const loadingSpinner = screen.getByTitle(loadingSpinnerTitle);
+    expect(loadingSpinner).toBeInTheDocument();
+
+    await waitForElementToBeRemoved(() => screen.queryByTitle(loadingSpinnerTitle));
+  });
+
+  it('should render field set', async () => {
+    renderEditBinding({});
+
+    await waitForElementToBeRemoved(() =>
+      screen.queryByTitle(textMock('ux_editor.modal_properties_loading')),
+    );
+
+    const fieldSet = screen.getByRole('group', { name: defaultEditBinding.label });
+    expect(fieldSet).toBeInTheDocument();
+  });
+
+  it('should render correct elements in field set', async () => {
+    renderEditBinding({
+      queries: {
+        getAppMetadataModelIds: getAppMetadataModelIdsMock,
+        getDataModelMetadata: getDataModelMetadataMock,
+      },
+    });
+
+    await waitForElementToBeRemoved(() =>
+      screen.queryByTitle(textMock('ux_editor.modal_properties_loading')),
+    );
+
+    const label = screen.getByText(defaultEditBinding.label);
+    expect(label).toBeInTheDocument();
+
+    const selectedModel = screen.getByText(defaultDataModel);
+    expect(selectedModel).toBeInTheDocument();
```

Comment:
hm.. for this part when the feature flag is not enabled, it will render the `StudioDisplayTile`. So it is not a disabled selector, but I could check that is not a combobox/selector? Like this below (it testing that it does exist when feature flag is enabled)
Hm... When the feature flag is not enabled, it will render the `StudioDisplayTile`, which is only text. I could check that it is not a combobox/selector, like this (for the feature flag enabled block, it tests the same but then it should exist):

```
    const selectedModel = screen.getByText(defaultDataModel);
    const selectedModelNotSelector = screen.queryByRole('combobox', {
      name: textMock('ux_editor.modal_properties_data_model_binding'),
    });
    expect(selectedModel).toBeInTheDocument();
    expect(selectedModelNotSelector).not.toBeInTheDocument();
  ```

Reply:
Hm... When the feature flag is not enabled, it will render the StudioDisplayTile, which is only text. I could check that it is not a combobox/selector, like this below (for the feature flag enabled block, it tests the same but then it should exist):

```
const selectedModel = screen.getByText(defaultDataModel);
const selectedModelNotSelector = screen.queryByRole('combobox', {
  name: textMock('ux_editor.modal_properties_data_model_binding'),
});
expect(selectedModel).toBeInTheDocument();
expect(selectedModelNotSelector).not.toBeInTheDocument();
```

Reply:
Doesn't seem like it after investigated the code and the html that displays in the test. The alternative could be to check if `data-testid="padlockIconTestId"` is available, that is set on the lock-icon.

Reply:
Ah okey, well, I guess it does not matter! Looks good like this 👏